tools/replace_code_block.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+status = 0
+
+
+def check_for_code(file: AnyStr, check: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param file: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    """
+    global status
+    with open(file, "rb") as infile:
+        content = infile.read()
+    old_content = content
+    new_content = content.replace(b".. code::", b".. code-block::")
+    if old_content != new_content:
+        if check:
+            status = 1
+            print(
+                "File {filename} contains ``code::`` statements!".format(
+                    filename=file
+                )
+            )
+        else:
+            with open(file, "wb") as outfile:
+                outfile.write(new_content)
+            print(
+                "Replaced ``code::`` statements to ``code-block::`` in "
+                "{filename}".format(filename=file)
+            )
+
+
+def main(check):
+    global status
+    for current_directory, folder_list, file_list in os.walk(
+        os.path.abspath(os.path.join(__file__, "..", "..", "praw"))
+    ):
+        for file in file_list:
+            file = "{dir}/{filename}".format(
+                dir=current_directory, filename=file
+            )
+            if file[-3:] == ".py":
+                try:
+                    check_for_code(file, check)
+                except Exception:
+                    print(
+                        "Error while checking file {filename}!".format(
+                            filename=file
+                        )
+                    )
+                    status = 2
+                    print_tb(sys.exc_info()[-1])
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Check and reformat praw's "
+        "usage of ``code::`` to "
+        "``code-block::``."
+    )
+    parser.add_argument(
```

Comment:
The "check" process should be the default, and as indicated above I think it'd make sense for the explicitly provided option name to be something like `overwrite` for the cases where we want this tool to fix the code.

Reply:
Black rewrites doe on default and that is why I designed the program this way

Reply:
> Black rewrites doe on default and that is why I designed the program this way

That's a good point. I guess the tool name is `replace_code_block` so I'm fine leaving it as is.